### PR TITLE
Send weights in registration

### PR DESCRIPTION
**Pull request by pawanbisht62**
This PR pertains to send extrinsics weights to parachain contract as each parachain will have different weights based on their pallet configuration.

Resolves: #90 

Depends on #94.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -597,20 +609,28 @@ pub mod pallet {
...
 		/// Registers the parachain with the Tellor controller contracts.
 		#[pallet::call_index(0)]
-		#[pallet::weight(115413000)]
+		#[pallet::weight(<T as Config>::WeightInfo::register())]
 		pub fn register(origin: OriginFor<T>) -> DispatchResult {
 			T::RegisterOrigin::ensure_origin(origin)?;
 			// Register with parachain registry contract
 			let registry_contract = T::Registry::get();
 			const GAS_LIMIT: u64 = gas_limits::REGISTER;
+			let weights = Weights {
```
Comment (evilrobot-01) [inline src/lib.rs:626]: Can we add this to the RegistrationSent event? Will be easiest way to see the actual weight values sent.

File: src/benchmarking.rs [hunk 2]
```diff
@@ -16,21 +16,640 @@
...
+			uint_value::<T>(6 * 10u128.pow(18)),
+			0,
+			staking_to_local_token_query_data)?;
+
+		T::BenchmarkHelper::set_time(12 * HOURS);
+	}: _(RawOrigin::Signed(reporter))
+
+	tip {
+		let s in 2..T::MaxTimestamps::get();
+		let query_data: QueryDataOf<T> = spot_price("dot", "usd").try_into().unwrap();
```
Comment (oraclown) [inline src/benchmarking.rs:282]: It seems `query_data` is defined and used a quite a few times in loops in this file for benchmarking, so you might consider varying the query data to be larger and smaller than just spot prices. Some other types [here](https://github.com/tellor-io/dataSpecs/tree/main/types), or you could just generate them randomly within bounds of query data max size.

File: .DS_Store
Comment (oraclown) [inline .DS_Store:1]: I think this Mac-specific file got uploaded accidentally. Might want to add to `.gitignore`?

**Review by oraclown — COMMENTED** [review]
Left a suggestion Tim had in our call. Thanks for this, @pawanbisht62

File: src/lib.rs (hunk 1 above)
Comment (pawanbisht62) [reply src/lib.rs:626]: Sure, thanks

File: .DS_Store
Comment (pawanbisht62) [reply .DS_Store:1]: Thanks for pointing out.

File: src/benchmarking.rs (hunk 2 above)
Comment (pawanbisht62) [reply src/benchmarking.rs:282]: Thanks for the suggestion, will resolve this comment in #94, as this PR is specific to sending weights to parachain contract.

Will clean this PR after merging #94 into main.

Comment (evilrobot-01) [reply src/benchmarking.rs:282]: Sounds like querydata should always be max length based on chain config to generate the worst case weight and then benchmark should take in actual query data length so caller can be refunded at runtime.

**Comment by pawanbisht62** [comment]
Closing this one as it creates some confusion around commit and some unwanted changes. Here is the new [PR](https://github.com/tellor-io/tellor-pallet/pull/104)

**Pull request closed without merging**